ui/selectors/multichain.ts

```diff
@@ -111,17 +123,55 @@ export function getMultichainNetwork(
 ): MultichainNetwork {
   const isEvm = getMultichainIsEvm(state, account);
 
-  // EVM networks
-  const evmNetworks: ProviderConfig[] = getAllNetworks(state);
-  const evmChainId: ChainId = getCurrentChainId(state);
-
   if (isEvm) {
-    const evmNetwork: ProviderConfig =
-      evmNetworks.find((provider) => provider.chainId === evmChainId) ??
-      getProviderConfig(state); // We fallback to the original selector otherwise
+    // EVM networks
+    const evmChainId: string = getCurrentChainId(state);
+
+    // TODO: Update to use network configurations when @metamask/network-controller is updated to 20.0.0
+    // ProviderConfig will be deprecated to use NetworkConfigurations
+    // When a user updates a network name its only updated in the NetworkConfigurations.
+    const evmNetwork: ProviderConfigWithImageUrlAndExplorerUrl =
+      getProviderConfig(state);
+    // Could be undefined for default configurations.
+    const evmNetworkConfigurations = getNetworkConfigurations(state);
+    const evmNetworkConfiguration =
+      // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+      evmNetworkConfigurations?.[evmNetwork.id!];
```

Comment:
Could we move this into this code branch:

```ts
if (evmNetwork.type === NETWORK_TYPES.RPC) {
  // Here
  
  ...
}
```

Also, I guess that custom networks always requires an `id` to be defined? If yes, we can also add a custom to justify the "no-non-null-assertion" in this case, WDYT?